# Project defiant

![some other image](https://github.com/PROJECT-DEFIANT/Project-defiant/blob/main/images/bioinfo.jpg?raw=true)

## Introduction

My name is Szymon and this is my story.

### Childhood

I was born in 1995 in a small town located north-east of Warsaw in Poland. I had pretty much regular childhood with no exceptions. As a nerdish person, I struggled over my time at highschool, I was not good enough in sports and had some fatal love crushes as most of the teenagers have. During that time I did not know what to do in my life. I went through a period of depression and resignation of my life.

### Adolescence

After highschool I have started chemistry studies at the Warsaw Military University of Technology, which caused me to run between my home and university wasting much of my free time. With unsuccessful exam session I decided to quit and get a job in nearby fast food chain restaurant. It was hard as I felt that I did not belong there, but still I needed to move on.

### Studies

During later summer of 2015 I started studies at Lodz University of Technology in a department of biotechnology. Which after 5 years I have graduated from with an engineer's diploma in technical biochemistry.

An important experience that I acquired was also internship in at the Intercollegiate Faculty of Biochemistry UG&MUG in Gdansk. I dealt with the process of purification of heat shock proteins sHsp in E.Coli bacteria. It required from me **good time management skills**, I needed to be **well organized** as well.

In my final year (2019) at the university I started learning bioinformatics for my diploma. The title of my thesis was "Short read _De novo_ assembling of plasmid sequences of Komagateibacter _xylinus_. I have chosen this topic to increase my competences in a field of genomics and to show myself that new challenges are not a problem for me. During my thesis I had to learn theoretical background of DNA sequencing, using bioinformatic tools like [IGV](https://software.broadinstitute.org/software/igv/), [SPAdes assembler](https://cab.spbu.ru/software/spades/), [Quast](https://quast.sourceforge.net/index.html) and learn linux, bash and python basics. This inspired me to continue my journey in a field of bioinformatics. I did not finish my master thesis, as staying at the university I would not learn practical knowledge, so I decided to move on yet again.

### Back to origins

After the studies I have moved back to my place of origin. I have started working in [Children's Health Memorial Institute in Warsaw](https://nauka.czd.pl/en/) as a junior assistant biotechnologist in genetics department. My main tasks was to uncover genetic background of rare disease occurring in patients DNA in polish population. This enabled me to understand the **rare genetic variant classification** based on ACMG guidelines as well as skills in usage of NCBI [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/), [SNP](https://www.ncbi.nlm.nih.gov/snp/), [OMIM](https://www.omim.org/), [OrphaNet](https://www.orpha.net/consor/cgi-bin/index.php), [HGMD](https://www.hgmd.cf.ac.uk/ac/index.php), [LOVD](https://www.lovd.nl/), [UniProt](https://www.uniprot.org/), [gnomAD](https://gnomad.broadinstitute.org/), [Reactome](https://reactome.org/), [HPO](https://hpo.jax.org/) and other genetic databases. In my day-to-day job I was analyzing BAM and VCF files coming from NGS target sequencing as well as from Whole exome sequencing, [fastqc](https://www.bioinformatics.babraham.ac.uk/projects/fastqc/) statistics and cnv variants from [CNVkit](https://cnvkit.readthedocs.io/en/stable/). As an aspiring bioinformatician I was responsible for meritorious background of my team of geneticists and lab analysts in the method of target enrichment sequencing.

I was also responsible for writing a grant for IT infrastructure to the Ministry of Education and Science, which obtained financing for purchase of a disk array and computing servers for in house genomics platform. During that period I had learned how to work with clients (as we were a laboratory we had to deal with patients and doctors constantly).

### Still learning

In late 2019 I had started postgraduate studies in **OMICS Data Science** - course about large scale sequencing data analysis in [Interdisciplinary Centre of Mathematical and Computational Modelling at University of Warsaw](https://icm.edu.pl/). This is a time when I also started [HEAP Metagenomics Pipelines](https://heap-exposome.eu/) internship in the same location.
During that period I have started learning how to code and refreshed my knowledge in python programming language. After the course I was able to code in python and R, started programming in C and pyspark as well as working oh high performance computing cluster with lustre filesystem and slurm queue task execution plan (Okeanos). I had also learned about the multiomics and how to apply bioconductor packages to miRNA and epigenomics analysis. I had my first contact with machine learning and deep learning in the matter of metagenomics analysis and mnist classifier.

### The MNM period

I was fortunate enough to work with unique and amazing professionals in MNM diagnostics by remote. This was a total game changer for me, as I developed my skills in **package development in R and python**, learn many technologies like **docker, git, aws s3 storage, seaborn for visualization, dvc**, automatic testing with **pytest**, using **flask** to host documentation websites, Writing **nextflow pipelines**. The biggest challenge was to use **pyspark to increase the efficiency in processing of genomic vcf files with the help of deltalake and glow package**. In MNM I used to work in **test driven development**, for day to day tasks I was using **slack (communication), github (codebase), google docs and sheets (documentation), jira (time & task management)**. I was responsible for developing **tools for analysis of the Whole Genome Sequencing data**, developing **pipeline for Panel sequencing** and packages for **extracting genomic features from tumor descriptors**. In MNM I had a chance to boost my software developer skills in many areas, beginning from how to handle large projects, working with **cloud native kubernetes solutions**, how to structure and maintain codebase keeping in mind rapidly changing business requirements for a startup.

That led me to start my own business, which I am currently developing - **Project Defiant**, where I want to utilize all my knowledge from science (genomics) and tech (software development) to achieve great things by combining both areas.

### Project Defiant

Right now I am focused on developing my brand, portfolio and contact network, to increase the ability to create impactful projects!

### Still learning v2

During the MNM period I was constantly busting my skills by learning new programming languages (Rust, Scala), **web development** and more. The whole list of technologies that I am familiar with can be found in the timeline section.


### QuartzBio

Right now I am a software develper at QuartzBio, part of Precision For Medicine, where I am working under the Virtual Sample Inventory project. The project is based
mainy in R language, I am gaininig new experience in maintaining (mostly debugging) and refactoring old code along with scheduling new solutions and expanding the test suite.
### For business

If my story sounds interesting to You and You want to cooperate in some area, feel free to email me at **szymonszyszkowski@gmail.com**

### For future

This was Project Defiant @ Szymon Szyszkowski

I am going to do some great stuff in the future and put it in here, so be sure to check out my blog for the next posts, otherwise Cześć !
